gadget/update.go

```diff
@@ -157,12 +157,22 @@ func searchForVolumeWithTraits(laidOutVol *LaidOutVolume, traits DiskVolumeDevic
 			AllowImplicitSystemData:          validateOpts.AllowImplicitSystemData,
 			ExpectedStructureEncryption:      validateOpts.ExpectedStructureEncryption,
 		}
-		ensureErr := EnsureVolumeCompatibility(laidOutVol.Volume, diskLayout, opts)
+		gadgetStructToDiskStruct, ensureErr := EnsureVolumeCompatibility(laidOutVol.Volume, diskLayout, opts)
 		if ensureErr != nil {
 			logger.Debugf("candidate disk %s not appropriate for volume %s due to incompatibility: %v", candidate.KernelDeviceNode(), vol.Name, ensureErr)
 			return false
 		}
 
+		// Set OnDiskStructure for laidOutVol, now that we know the exact match
```

Comment:
I wonder now that it does this whether this function needs a slightly different name and a doc comment, also what is the relation of this and gadget/install package applyOnDiskStructureToLaidOut ?